tests/datasets/test_datasets.py

```diff
@@ -1,41 +1,144 @@
 import pytest
 import torch
-from torch.utils.data import DataLoader
+from torch.utils.data import DataLoader, Dataset
 
 from pl_bolts.datasets import DummyDataset, RandomDataset, RandomDictDataset, RandomDictStringDataset
+from pl_bolts.datasets.dummy_dataset import DummyDetectionDataset
 from pl_bolts.datasets.sr_mnist_dataset import SRMNIST
 
 
-def test_dummy_ds():
-    ds = DummyDataset((1, 2), num_samples=100)
-    dl = DataLoader(ds)
+@pytest.mark.parametrize("batch_size,num_samples", [(16, 100), (1, 0)])
+def test_dummy_ds(catch_warnings, batch_size, num_samples):
 
-    for b in dl:
-        pass
+    if num_samples > 0:
 
+        ds = DummyDataset((1, 28, 28), (1,), num_samples=num_samples)
+        dl = DataLoader(ds, batch_size=batch_size)
 
-def test_rand_ds():
-    ds = RandomDataset(32, num_samples=100)
-    dl = DataLoader(ds)
+        assert isinstance(ds, Dataset)
+        assert num_samples == len(ds)
 
-    for b in dl:
-        pass
+        x = next(iter(ds))
+        assert x[0].shape == torch.Size([1, 28, 28])
+        assert x[1].shape == torch.Size([1])
 
+        batch = next(iter(dl))
+        assert batch[0].shape == torch.Size([batch_size, 1, 28, 28])
+        assert batch[1].shape == torch.Size([batch_size, 1])
 
-def test_rand_dict_ds():
-    ds = RandomDictDataset(32, num_samples=100)
-    dl = DataLoader(ds)
+    else:
+        with pytest.raises(Exception) as e:
```

Comment:
We can expect specific exception type (`ValueError`) and match the string of exception in this context manager

Reply:
This applies to all other tests

Reply:
@otaj . sorry I didn't understand what this means exactly? I mean I have done exactly that in the next two `assert` statements. Did you mean that the assert statements should be inside the context manager as well?

Reply:
```suggestion
        with pytest.raises(ValueError, match="regexp to match the error text"):
```
and you can omit those two `asserts` then :rocket:

Reply:
made the commit with the necessary changes.thanks!